File: CPP/class/const/more/p1.cpp
#include<iostream>
using namespace std;
class Demo{
	int x = 10;
	public:
		Demo(){
			cout<<"constructor"<<endl;
		}


		 void fun()const {
			cout<<++x<<endl;//x cannot be modified because x belong to object of class

		}



}
;
int main(){
	Demo obj;




}

File: java/static blocks/p1.java
class Demo{
    
    static int a =10;

    static{
        System.out.println("static 1");
    }

    {
        System.out.println("instance 1");

    }

    static{
        System.out.println("static 2");

    }

    Demo(){
          System.out.println("constructor");

    }


    {
        System.out.println("instance 2");

    }

    static void fun(){

    }

}




class Test{
    public static void main(String args[]){

        Demo d1 = new Demo();
        Demo d2 = new Demo();
      


        Demo.fun();
    //    Demo.fun();

    }
}